js/util/extractProps.ts

```diff
@@ -0,0 +1,47 @@
+const pick = require('lodash.pick');
+const flatten = require('flat');
+
+export interface Component {
+  //TODO: use the real RN ones
+  props: {
+    [propertyKey: string]: any;
+  };
+}
+
+export interface PropExtractorCriteria {
+  include: string[];
+  exclude: string[];
+}
+
+export interface PropExtractorConfig {
+  [componentName: string]: PropExtractorCriteria;
+}
+
+export const extractProps = (
+  elementName: string,
+  component: Component,
+  config: PropExtractorConfig
+): string => {
+  if (!component || !config[elementName]) {
```

Comment:
Do we need the `!component`? I'd think that wouldn't be necessary unless `component` was of type `Component | null` or `Component?`.

Reply:
Sure, but I'm trying to be extra careful that this code won't crash a running app.  Typescript only gives you compile time checks - I still worry about the runtime.

Reply:
If there's a chance that `!component` could be `true`, the type should probably be changed to `Component | null` or `Component?`.